mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/repositories/DeltakerRepository.kt

```diff
@@ -0,0 +1,91 @@
+package no.nav.mulighetsrommet.api.repositories
+
+import kotliquery.Row
+import kotliquery.queryOf
+import no.nav.mulighetsrommet.api.utils.DatabaseUtils
+import no.nav.mulighetsrommet.database.Database
+import no.nav.mulighetsrommet.domain.dbo.DeltakerDbo
+import no.nav.mulighetsrommet.domain.dbo.Deltakerstatus
+import org.intellij.lang.annotations.Language
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class DeltakerRepository(private val db: Database) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun upsert(deltaker: DeltakerDbo): DeltakerDbo {
+        logger.info("Lagrer deltaker id=${deltaker.id}")
+
+        @Language("PostgreSQL")
+        val query = """
+            insert into deltaker (id, tiltaksgjennomforing_id, norsk_ident, status, start_dato, slutt_dato, registrert_dato)
+            values (:id::uuid, :tiltaksgjennomforing_id::uuid, :norsk_ident, :status::deltakerstatus, :start_dato, :slutt_dato, :registrert_dato)
+            on conflict (id)
+                do update set tiltaksgjennomforing_id = excluded.tiltaksgjennomforing_id,
+                              status                  = excluded.status,
+                              norsk_ident             = excluded.norsk_ident,
+                              start_dato              = excluded.start_dato,
+                              slutt_dato              = excluded.slutt_dato,
+                              registrert_dato         = excluded.registrert_dato
+            returning *
+        """.trimIndent()
+
+        return queryOf(query, deltaker.toSqlParameters())
+            .map { it.toDeltakerDbo() }
+            .asSingle
+            .let { db.run(it)!! }
+    }
+
+    fun getAll(tiltaksgjennomforingId: UUID? = null): List<DeltakerDbo> {
+        val where = DatabaseUtils.andWhereParameterNotNull(
+            tiltaksgjennomforingId to "tiltaksgjennomforing_id = :tiltaksgjennomforing_id"
```

Comment:
Skulle `:tiltaksgjennomforing_id` hatt `::uuid` bak seg slik `:tiltaksgjennomforing_id::uuid`?

Reply:
Bra spotta! La till en testsuite for klassen også